glogger/extra_adapter.py

```diff
@@ -39,6 +46,17 @@ def process(self, msg: Any, kwargs: MutableMapping[str, Any]) -> Tuple[Any, Muta
             logging_kwargs["extra"] = extra
 
         extra = self.get_extra(**logging_kwargs)
+
+        if logging_kwargs.get("exc_info") is True:
+            # If exc_info is True, and the exception is subclassing ExtraException, then add the extra attributes
+            # from the exception to the extra attributes of the record:
+            exc_info = sys.exc_info()
+            # If exc_info is True, then there must be an exception:
+            assert exc_info is not None
+            if isinstance(exc_info[1], ExtraException):
```

Comment:
Don't actually have to require it. I don't think logging does. Also, I don't want to derive from `ExtraException` everywhere now. It's easier to just support it transparently by checking for `.extra`, or even better can just `extra = e.__dict__`
```suggestion
            if exc_info and hasattr(exc_info[1], 'extra'):
```

Reply:
I don't think we want it implicit (i.e ALWAYS take the `__dict__` of an exception). I agree it's nicer without requiring to subclass `ExtraException` tho.

Reply:
So I'm keeping the base class for syntactic sugar and having the adapter doing getattr("extra"), aye?